task/zap-baseline/0.1/zap-baseline.yaml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: zap-baseline
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: cli
+    tekton.dev/displayName: "OWASP Zap Baseline"
+spec:
+  description: >-
+    OWASP Zap Baseline is a command line tool,
+    testing api endpoints for vulnerabilities.
+
+  workspaces:
+    - name: zap-folder
+      description: A RW folder where all zap output files is stored
+      mountPath: /zap/wrk
+  params:
+    - name: testURL
+      description: URL of to be tested, must include https:// or http://
+      type: string
+      default: ""
+    - name: configFile
+      description: Path to zap-scan-config
+      type: string
+      default: scan.conf
+    - name: cmName
+      description: Name of cm containing scan.conf
+      type: string
+      default: zap-scan-config
+    - name: mins
+      description: The number of minutes to spider for
+      type: string
+      default: "1"
+  steps:
+    - name: zap-baseline-scan
+      image: quay.io/nissessenap/zap-proxy:0.0.1
+      command:
+        - zap-baseline.py
+      args:
+        - -t
+        - $(params.testURL)
+        - -c
+        - $(params.configFile)
+        - -m
+        - $(params.mins)
+        - -w
+        - report.md
+      volumeMounts:
+        - name: zap-scan-config
+          mountPath: "/zap/wrk/$(params.configFile)"
+          subPath: "$(params.configFile)"
+
+
+  volumes:
+    - name: zap-scan-config
+      configMap:
+        name: "$(params.cmName)"
```

Comment:
Sure I will read up on how it's done and update the PR.

Reply:
Got it to work by updating task with:
```
  workspaces:
    - name: zap-scan-config
      mountPath: "/zap/wrk/scan.conf"

    - name: zap-folder
      description: A RW folder where all zap output files is stored
      mountPath: /zap/wrk
```

And the taskRun with:
```
  workspaces:
    - name: zap-folder
      persistentVolumeClaim:
        claimName: zap-report
    - name: zap-scan-config
      configmap:
        name: zap-scan-config
      subPath: scan.conf
```

The boring part is that the current version of zap uses /zap/wrk as the basedir so that's why I have to do subPath stuff.

It's kind of boring to write /zap/wrk in two locations. Is it possible to ether use: $(workspaces.<name>.path) in another workspace or in a param?
Tried a bit back and forth but didn't get it to work.